sdk/core/azure-core/azure/core/pipeline/transport/base.py

```diff
@@ -298,6 +335,51 @@ def stream_download(self, pipeline):
         """
 
 
+class HttpClientTransportResponse(HttpResponse):
+    """Create a HTTPResponse from an http.client response.
+
+    Body will NOT be read by the constructor. Call "body()" to load the body in memory if necessary.
+
+    :param HttpRequest request: The request.
+    :param httpclient_response: The object returned from an HTTP(S)Connection from http.client
+    """
+    def __init__(self, request, httpclient_response):
+        super(HttpClientTransportResponse, self).__init__(request, httpclient_response)
+        self.status_code = httpclient_response.status
+        self.headers = dict(httpclient_response.getheaders())
```

Comment:
Any problems with case sensitivity?

Reply:
Yes, that should be case-insitive, to respect the contract of HttpResponse, even if here that's not critical

Reply:
Thinking about, this class will be the one returned as parts, so it's important that it's case-insensitive, need to fix it

Reply:
Fixed